flocker/node/agents/test/test_blockdevice.py

```diff
@@ -391,32 +549,74 @@ def assert_discovered_state(case,
     )
     discovering = deployer.discover_state(previous_state)
     local_state = case.successResultOf(discovering)
-    expected_paths = {}
-    for manifestation in expected_manifestations:
-        dataset_id = manifestation.dataset.dataset_id
-        mountpath = deployer._mountpath_for_manifestation(manifestation)
-        expected_paths[dataset_id] = mountpath
-    case.assertEqual(NodeState(
-        applications=None,
-        uuid=deployer.node_uuid,
-        hostname=deployer.hostname,
-        manifestations={
-            m.dataset_id: m for m in expected_manifestations},
-        paths=expected_paths,
-        devices=expected_devices,
-    ), local_state.node_state
-    )
-    # FLOC-1806 - Make this actually be a dictionary (callers pass a list
-    # instead) and construct the ``NonManifestDatasets`` with the
-    # ``Dataset`` instances that are present as values.
+
     case.assertEqual(
-        NonManifestDatasets(datasets={
-            unicode(dataset_id):
-            Dataset(dataset_id=unicode(dataset_id))
-            for dataset_id in expected_nonmanifest_datasets
-        }), local_state.nonmanifest_datasets)
+        local_state,
+        BlockDeviceDeployerLocalState(
+            hostname=deployer.hostname,
+            node_uuid=deployer.node_uuid,
+            datasets={
+                dataset.dataset_id: dataset
+                for dataset in expected_discoved_datasets
+            },
+            volumes=expected_volumes,
+        )
+    )
+
+
+class BlockDeviceDeployerDiscoverRawStateTests(SynchronousTestCase):
+    """
+    Tests for ``BlockDeviceDeployer._discover_raw_state``.
```

Comment:
I'm not sure how much (if any) tests the warrants.

One difficulty with testing this, is that it reports state that isn't under our control as well.